lib/teiserver/microblog/schemas/poll_response.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Teiserver.Microblog.PollResponse do
+  @moduledoc false
+  use TeiserverWeb, :schema
+
```

Comment:
This module should ideally define the types:
```elixir
  @type id :: Ecto.UUID.t()
  @type t :: %__MODULE__{
          id: id(),
          uploader: Teiserver.Account.User.t(),
          filename: String.t(),
          type: String.t(),
          file_size: integer(),
          inserted_at: DateTime.t(),
          updated_at: DateTime.t()
        }
```

This defines the type `Upload.id()`, and correctly map it to an ecto uuid. This also fixes the error you were having with dialyzer and `delete_upload`. Because it was expecting a `Ecto.UUID.t()` but was receiving a `Upload.id()` and it's not smart enough without the additional type definition that these things are the same.

Reply:
Explains the error perfectly, thank you.